sdk/webpubsub/azure-messaging-webpubsubclient/azure/messaging/webpubsubclient/aio/_client.py

```diff
@@ -0,0 +1,1179 @@
+        message_retry_backoff_factor: float = _RETRY_BACKOFF_FACTOR,
+        message_retry_backoff_max: float = _RETRY_BACKOFF_MAX,
+        message_retry_mode: RetryMode = RetryMode.Exponential,
+        message_retry_total: int = _RETRY_TOTAL,
+        protocol_type: WebPubSubProtocolType = WebPubSubProtocolType.JSON_RELIABLE,
+        reconnect_retry_backoff_factor: float = _RETRY_BACKOFF_FACTOR,
+        reconnect_retry_backoff_max: float = _RETRY_BACKOFF_MAX,
+        reconnect_retry_mode: RetryMode = RetryMode.Exponential,
+        reconnect_retry_total: int = _RETRY_TOTAL,
+        auto_rejoin_groups: bool = True,
+        logging_enable: bool = False,
+        ack_timeout: float = _ACK_TIMEOUT,
+        start_timeout: float = _START_TIMEOUT,
+        user_agent: Optional[str] = None,
+        **kwargs: Any,
+    ) -> None:
+        if isinstance(credential, WebPubSubClientCredential):
+            self._credential = credential
+        elif isinstance(credential, str):
+            self._credential = WebPubSubClientCredential(credential)
+        else:
+            raise TypeError("type of credential must be str or WebPubSubClientCredential")
+        super().__init__(
+            message_retry_backoff_factor=message_retry_backoff_factor,
+            message_retry_backoff_max=message_retry_backoff_max,
+            message_retry_mode=message_retry_mode,
+            message_retry_total=message_retry_total,
+            protocol_type=protocol_type,
+            reconnect_retry_backoff_factor=reconnect_retry_backoff_factor,
+            reconnect_retry_backoff_max=reconnect_retry_backoff_max,
+            reconnect_retry_mode=reconnect_retry_mode,
+            reconnect_retry_total=reconnect_retry_total,
+            auto_rejoin_groups=auto_rejoin_groups,
+            logging_enable=logging_enable,
+            ack_timeout=ack_timeout,
+            start_timeout=start_timeout,
+            user_agent=user_agent,
+            **kwargs,
+        )
+        self._ack_map: AckMapAsync = AckMapAsync()
+        self._ws: Optional[WebSocketAppAsync] = None
+        self._event: asyncio.Event = asyncio.Event()
+        self._task_seq_ack: Optional[asyncio.Task] = None
+        self._task_run_forever: Optional[asyncio.Task] = None
+
+    def _next_ack_id(self) -> int:
+        self._ack_id = self._ack_id + 1
+        return self._ack_id
+
+    async def _send_message(self, message: WebPubSubMessage, **kwargs: Any) -> None:
+        pay_load = self._protocol.write_message(message)
+        if self._ws.closed:  # type: ignore
+            raise SendMessageError("The websocket connection is not connected.")
+
+        await self._ws.send(pay_load)  # type: ignore
+        if kwargs.pop("logging_enable", False) or self._logging_enable:
+            _LOGGER.debug("\nconnection_id: %s\npay_load: %s", self._connection_id, pay_load)
+
+    async def _send_message_with_ack_id(
+        self,
+        message_provider: Callable[[int], WebPubSubMessage],
```

Comment:
I'm not sure why we're using this lambda/callable structure, when all of the Message types just have an optional attribute called `ack_id`..... could we not simplify it by doing:
```python
async def _send_message_with_ack_id(message: WebPubSubMessage, **kwargs):
    if message.ack_id is None:
        message.ack_id = self._next_ack_id()
```

Reply:
Make sense and will update.